src/hole.rs

```diff
@@ -271,17 +271,21 @@ fn deallocate(mut hole: &mut Hole, addr: usize, mut size: usize) {
 
                 // or: this is the last hole
                 // before:  ___XXX_________    where X is this hole
-                // after:   ___XXX__FFFF___    where F is the freed block
-
-                let new_hole = Hole {
-                    size: size,
-                    next: hole.next.take(), // the reference to the Y block (if it exists)
-                };
-                // write the new hole to the freed memory
-                let ptr = addr as *mut Hole;
-                mem::forget(mem::replace(unsafe { &mut *ptr }, new_hole));
-                // add the F block as the next block of the X block
-                hole.next = Some(unsafe { Unique::new(ptr) });
+                if hole_addr + hole.size == addr {
+                    // after:   ___XXXFFFF___    where F is the freed block
+                    hole.size += size;
```

Comment:
Good catch! I think we could merge this case with [lines 243-249](https://github.com/rafalmiel/linked-list-allocator/blob/b7dc99795964afed5a2ab64f6bc8d6c6e193e89e/src/hole.rs#L243-L249) by replacing the `Some(_)` with a `None`. Or am I missing something?